src/linalg/spectral.jl

```diff
@@ -151,13 +151,22 @@ function incidence_matrix(g::AbstractGraph, T::DataType=Int; oriented=false)
     # every col has the same 2 entries
     colpt = collect(1:2:(nz + 1))
     nzval = repeat([(isdir || oriented) ? -one(T) : one(T), one(T)], n_e)
+    swap(i,j) = begin
+        tmp = nzval[i]
+        nzval[i] = nzval[j]
+        nzval[j] = tmp
+    end
 
     # iterate over edges for row indices
     rowval = zeros(Int, nz)
     i = 1
-    for u in vertices(g)
-        for v in outneighbors(g, u)
+    for u in sort(vertices(g))
+        for v in sort(outneighbors(g, u))
```

Comment:
`outneighbors` is similarly sorted.

Reply:
If outneighbors is already sorted then this bug shouldn't happen? unless it is only the v<u case that causes the problem?

Reply:
`outneighbors` is definitely already sorted, since the `fadjlist` is created in sorted order.